interaction draft

tgerke opened the pull request:
draft of interaction and effect modification chapter

tgerke committed:
interaction draft

malcolmbarrett commented:
Another couple of things I'd like to see in this chapter:

1) we need an example from our dataset. Maybe ticket season * extra magic morning? I think a good way to look at this would be to keep your simulated walkthrough then put this at the end and consider the question from the functional form vs. effect modification vs. joint cause approach. Like take the same model and consider how we would use it for all three approaches depending on our opinion on the structure/causal question
2) At this point, we have covered g-computation, too, so related to @LucyMcGowan's chapter, I'd like to see a discussion of that (also related to my marginaleffects comment)

malcolmbarrett commented on chapters/14-interaction.qmd:
```diff
@@ -5,13 +5,689 @@
+
+### Marginal structural models
+
+**Marginal structural models (MSMs)** provide a framework for distinguishing between effect modification and interaction in the analysis [@VanderWeele2009].
```
I feel like we need a callout box for the term marginal structural models which is used to refer to many techniques, presumably because it is a generalization of them (but I haven't thought it through or looked into usage all that much)

malcolmbarrett commented on chapters/14-interaction.qmd:
```diff
@@ -5,13 +5,689 @@
+
+For interaction, we need weights for both E and Q:
+$$w^E = \frac{P(E = e | Q = q)}{P(E = e | Q = q, X = x)}$$
+$$w^Q = \frac{P(Q = q)}{P(Q = q | X = x)}$$
```
I think we need some intuition here, given the symmetric nature of interaction you lay out, about why we have 

$$w^Q = \frac{P(Q = q)}{P(Q = q | X = x)}$$

and not 

$$w^Q = \frac{P(Q = q)}{P(Q = q | E = e, X = x)}$$

malcolmbarrett reviewed (changes requested):
thanks for getting this in! lots of good, nuanced content here. 

I've requested some additional changes in addition to my comments in the thread

malcolmbarrett commented:
Having slept on this, here are a few more things I wanted to include in this chapter:

- Discuss the relationship between heterogeneous effects and the ATT+ ATU
- Relatedly, use (D'Agostino)^3's method for visualizing heterogeneous effects: https://github.com/r-causal/halfmoon/issues/2
- Include more visualizations to improve intuition. Check out https://marginaleffects.com/chapters/interactions.html and https://bsky.app/profile/dingdingpeng.the100.ci/post/3louqbmtwrk2u
- Tighten up the end of the chapter. It feels like there is some empty calories here.

Reviewing my notes on this chapter, I also found https://www.the100.ci/2024/06/26/sometimes-a-causal-effect-is-just-a-causal-effect-regardless-of-how-its-mediated-or-moderated/ which you might find as an interesting perspective

LucyMcGowan commented on chapters/14-interaction.qmd:
```diff
@@ -5,13 +5,689 @@
+
+## Effect modification versus interaction
+
+**Effect modification** occurs when the effect of one exposure (E) on an outcome (D) varies across strata of a third variable (Q).
```
```suggestion
**Effect modification** occurs when the effect of one exposure (X) on an outcome (Y) varies across strata of a third variable (Q).
```

LucyMcGowan commented on chapters/14-interaction.qmd:
```diff
@@ -5,13 +5,689 @@
+This definition asks: does the effect of E vary depending on what level we set Q to?
+
+Because interaction is symmetric, this definition is equivalent to asking whether the effect of Q varies depending on what level we set E to:
+$$E[D_{e_1 q_1}] - E[D_{e_1 q_0}] \neq E[D_{e_0 q_1}] - E[D_{e_0 q_0}]$$
```
```suggestion
$$E[Y(1, q_1)] - E[Y(1, q_0)] \neq E[Y(0, q_1)] - E[Y(0, q_0)]$$
```

LucyMcGowan commented on chapters/14-interaction.qmd:
```diff
@@ -5,13 +5,689 @@
+conventional_dag <- dagify(
+  E ~ X,
+  Q ~ X,
+  D ~ E + Q + X,
```
```suggestion
  Y ~ X + Q + Z,
```

LucyMcGowan commented on chapters/14-interaction.qmd:
```diff
@@ -5,13 +5,689 @@
+    )
+  ),
+  exposure = "E",
+  outcome = "D",
```
```suggestion
  outcome = "Y",
```

LucyMcGowan commented on chapters/14-interaction.qmd:
```diff
@@ -5,13 +5,689 @@
+  labels = c(
+    E = "Drug",
+    D = "Hypertension",
+    X = "Genotype",
```
```suggestion
    Z = "Genotype",
```

LucyMcGowan commented on chapters/14-interaction.qmd:
```diff
@@ -5,13 +5,689 @@
+  labels = c(
+    E = "Weight-loss\ndrug",
+    D = "Weight",
+    X = "Sugar\nintake",
```
```suggestion
    Z = "Sugar\nintake",
```

LucyMcGowan commented on chapters/14-interaction.qmd:
```diff
@@ -5,13 +5,689 @@
+  # Sugar intake (confounder)
+  X = rnorm(n),
+  # Exercise level (affected by sugar intake)
+  Q = rbinom(n, 1, plogis(-0.5 + 1.2 * X)),
```
```suggestion
  Q = rbinom(n, 1, plogis(-0.5 + 1.2 * Z)),
```

LucyMcGowan reviewed (commented):
Ok I started this review but I have general questions about terminology -- I don't think effect modification is necessarily in the absence of confounders? I think of effect modification just meaning is the effect modified by a particular variable (i.e. do we observe a different effect in different strata, these could be after conditioning on confounders) and interaction (as it seems like it is defined in the beginning of this chapter) to mean is it *causal* i.e. is Q actually the cause of the difference we see (vs. some other variable that is just correlated with Q) -- but then in the simulation it seems like we've moved away from those definitions (@tgerke do you mind weighing in?)

LucyMcGowan reviewed (commented):
Ok I started this review but I have general questions about terminology -- I don't think effect modification is necessarily in the absence of confounders? I think of effect modification just meaning is the effect modified by a particular variable (i.e. do we observe a different effect in different strata, these could be after conditioning on confounders) and interaction (as it seems like it is defined in the beginning of this chapter) to mean is it *causal* i.e. is Q actually the cause of the difference we see (vs. some other variable that is just correlated with Q) -- but then in the simulation it seems like we've moved away from those definitions (@tgerke do you mind weighing in?)

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Apply suggestions from code review

Co-authored-by: Lucy D'Agostino McGowan <LucyMcGowan@users.noreply.github.com>

LucyMcGowan committed:
Apply suggestions from code review

Co-authored-by: Lucy D'Agostino McGowan <LucyMcGowan@users.noreply.github.com>

LucyMcGowan committed:
Apply suggestions from code review

Co-authored-by: Lucy D'Agostino McGowan <LucyMcGowan@users.noreply.github.com>

LucyMcGowan committed:
Apply suggestions from code review

Co-authored-by: Lucy D'Agostino McGowan <LucyMcGowan@users.noreply.github.com>

LucyMcGowan committed:
Apply suggestions from code review

Co-authored-by: Lucy D'Agostino McGowan <LucyMcGowan@users.noreply.github.com>

LucyMcGowan committed:
Apply suggestions from code review

Co-authored-by: Lucy D'Agostino McGowan <LucyMcGowan@users.noreply.github.com>

LucyMcGowan committed:
Apply suggestions from code review

Co-authored-by: Lucy D'Agostino McGowan <LucyMcGowan@users.noreply.github.com>

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Update chapters/14-interaction.qmd

LucyMcGowan committed:
Apply suggestions from code review

Co-authored-by: Lucy D'Agostino McGowan <LucyMcGowan@users.noreply.github.com>

LucyMcGowan committed:
Apply suggestions from code review

Co-authored-by: Lucy D'Agostino McGowan <LucyMcGowan@users.noreply.github.com>

LucyMcGowan committed:
tie back to the estimands

LucyMcGowan committed:
why are they different? also need better word than things

LucyMcGowan committed:
add callout and explain interaction

LucyMcGowan committed:
add seven dwarfs example

LucyMcGowan committed:
update

LucyMcGowan committed:
add in IPW

LucyMcGowan committed:
tie in travis things

LucyMcGowan committed:
remove old

LucyMcGowan committed:
Dags

LucyMcGowan committed:
ref sec dags

LucyMcGowan committed:
reconcile

Merge branch 'interaction' of https://github.com/r-causal/causal-inference-in-R into interaction

# Conflicts:
#	chapters/14-interaction.qmd
